### ISSUE TITLE
toJSON should not include the link’s selectors

### ISSUE THREAD

**Issue by nomeata** [clientIO/joint#177]
`graph.toJSON` is supposed to be used for serialization and persistence. Nevertheless, I find that it contains code like

```
"target":{"id":"04a59ab3-eccb-417e-9ccd-4d855a056100","selector":"g:nth-child(1) > path:nth-child(3)","port":"in1"}
```

The content of the `selector` is not a model value, but rather an implementation detail of the `View`, and it worries me that I might not be able to load back a persisted graph after I have changed the my views for the element.

`joint.dia.link` contains a function `makeSelector` that recovers the selector from the `port`, it seems, so it would make sense to _not_ include `selector` in the serialization and use that in `fromJSON`.

**Comment by DavidDurman** [comment]
@nomeata good point! However, the thing is that one does not have to have a port defined but still want a link to be connected to a `'magnet'`, where `magnet` is in our terms any SVG element that a link can "stick" to. In other words, you can mark any SVG subelement of your shapes a `'magnet'` (by setting the 'magnet' attribute - http://jointjs.com/api#SpecialAttributes:magnet) which makes it possible to connect links to it. The magnet must somehow be identifiable in the link models so that the diagram can be reconstructed. Now we use `'selector'` with a complete (and therefore unique) path to that magnet.

One thing I'd like to point out is that it is in our roadmap for JointJS v1.0 to make it easy to work with ports in any shape. As part of that, we will most likely introduce a more robust and better way to identify ports/magnets.

**Comment by nomeata** [comment]
Ok, I was somehow expecting that the port attribute is optional and hence this behaviour.

Incidentially, I am right now writing my own custom View with custom port shapes, by overriding `renderMarkup`. It felt too wrong to have view-related data (`markup`, `attrs`) in the model...

**Comment by kumilingus** [comment]
Fixed in `v2.1.3`.
